src/util/validate.js

```diff
@@ -1,36 +1,41 @@
-const checkConstraint = (record, field, constraint) => {
-    if (typeof constraint.custom === 'function') {
-        return constraint.custom(record[field], record);
-    }
+/* @link http://stackoverflow.com/questions/46155/validate-email-address-in-javascript */
```

Comment:
Your regex is not valid since it refuse this email: `Kmaschta-Thinkpad@localhost` which is a correct one.

Here is the [100% correct way to validate email addresses](https://hackernoon.com/the-100-correct-way-to-validate-email-addresses-7c4818f24643#.py969fn84).
Check only the format `X@Y` or don't, it's a bad user experience.